File: packages/case-example-mock-types/src/states/InState.ts
import { SETUP_NAMED_STATE } from '@contract-case/case-entities-internal';
import { AnyState } from './AnyState';

/**
 * A state descriptor for configuring an example that needs to run in a
 * particular named state.
 *
 * @public
 */
export class InState extends AnyState {
  /** @internal */
  readonly '_case:state:type': typeof SETUP_NAMED_STATE;

  /**
   * Constructs a new state descriptor
   *
   * @param stateName - The name of the state used by this example. This must
   * match one of the state handlers provided in the configuration during the
   * example run.
   */
  constructor(stateName: string) {
    super(SETUP_NAMED_STATE, stateName);
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/states/InState.java
package io.contract_testing.contractcase.dsl.states;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslState;
import java.lang.String;
import javax.annotation.Generated;
import lombok.Builder;
import lombok.Getter;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * A state descriptor for configuring an example
 * that needs to run in a particular named state.
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class InState<M> implements DslState {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:state:type")
  private final String type;

  /**
   * The name of the state used by this example.
   * This must match one of the state handlers provided in the configuration during the
   * example run.
   */
  @Getter
  @JsonProperty("stateName")
  private final String stateName;

  @Builder
  public InState(@NotNull final String stateName) {
    this.type = "_case:NamedState";
    this.stateName = stateName;
  }
}
